### ISSUE TITLE
Wont start in Ubuntu 24.04 LTS

### ISSUE THREAD

**Issue by paulhiggs** [leon3s/wireguard-gui#140]
I installed wireguard and wireguard-cli but trying to launch wireguard-cli fails, both in user and administrator modes...
<pre><font color="#26A269"><b>paul@ENVY</b></font>:<font color="#12488B"><b>~</b></font>$ wireguard-gui
current: Err(Os { code: 2, kind: NotFound, message: &quot;No such file or directory&quot; })
payload: IpPayload { origin: &quot;188.74.74.72&quot; }
app inited
app state: AppSt(Mutex { data: &lt;locked&gt; })
Gtk-<font color="#26A269"><b>Message</b></font>: <font color="#12488B">02:52:08.854</font>: Not loading module &quot;atk-bridge&quot;: The functionality is provided by GTK natively. Please try to not load it.

(wireguard-gui:19106): Gtk-<font color="#A2734C"><b>WARNING</b></font> **: <font color="#12488B">02:52:08.935</font>: GTK+ module /snap/wireguard-gui/5/gnome-platform/usr/lib/gtk-2.0/modules/libcanberra-gtk-module.so cannot be loaded.
GTK+ 2.x symbols detected. Using GTK+ 2.x and GTK+ 3 in the same process is not supported.
Gtk-<font color="#26A269"><b>Message</b></font>: <font color="#12488B">02:52:08.935</font>: Failed to load module &quot;canberra-gtk-module&quot;

(wireguard-gui:19106): Gtk-<font color="#A2734C"><b>WARNING</b></font> **: <font color="#12488B">02:52:08.936</font>: GTK+ module /snap/wireguard-gui/5/gnome-platform/usr/lib/gtk-2.0/modules/libcanberra-gtk-module.so cannot be loaded.
GTK+ 2.x symbols detected. Using GTK+ 2.x and GTK+ 3 in the same process is not supported.
Gtk-<font color="#26A269"><b>Message</b></font>: <font color="#12488B">02:52:08.936</font>: Failed to load module &quot;canberra-gtk-module&quot;
libEGL fatal: DRI driver not from this Mesa build (&apos;23.2.1-1ubuntu3.1~22.04.2&apos; vs &apos;23.0.4-0ubuntu1~22.04.1&apos;)

<font color="#26A269"><b>paul@ENVY</b></font>:<font color="#12488B"><b>~</b></font>$ 
<font color="#26A269"><b>paul@ENVY</b></font>:<font color="#12488B"><b>~</b></font>$ 
<font color="#26A269"><b>paul@ENVY</b></font>:<font color="#12488B"><b>~</b></font>$ 
<font color="#26A269"><b>paul@ENVY</b></font>:<font color="#12488B"><b>~</b></font>$ sudo wireguard-gui
mkdir: cannot create directory ‘/run/user/0’: Permission denied
current: Err(Os { code: 2, kind: NotFound, message: &quot;No such file or directory&quot; })
payload: IpPayload { origin: &quot;188.74.74.72&quot; }
app inited
app state: AppSt(Mutex { data: &lt;locked&gt; })
Authorization required, but no authorization protocol specified

(wireguard-gui:19222): Gtk-<font color="#A2734C"><b>WARNING</b></font> **: <font color="#12488B">02:52:20.063</font>: cannot open display: :0
</pre>

Am i missing some prerequisite? I dont want to develop or build, just run it

**Comment by paulhiggs** [comment]
Note that `snap install wireguard-gui` installs 0.1.1

<pre><font color="#26A269"><b>paul@ENVY</b></font>:<font color="#12488B"><b>~</b></font>$ snap install wireguard-gui
wireguard-gui 0.1.1 from leone (leonev) installed
</pre>

**Comment by leon3s** [comment]
Hey thanks for the report, the snap version doesn't work properly currently.

The missing dependencies such as `libcanberra-gtk-module libjavascriptcoregtk-4.0-18 libicu70 libwebkit2gtk-4.0-37` is weird i didn't had to install them on a fresh install, i'll add them in the prerequise in the readme

**Comment by leon3s** [comment]
I have updated the readme, closing this issue.

**Comment by melroy89** [comment]
@paulhiggs so the deb file is broken for 24.04 you say? Since I don't have these issues either. However I installed the deb on Linux mint 21.3, which is actually based on a older Ubuntu: Ubuntu 22.04.

**Comment by melroy89** [comment]
This is the error I see when trying to install the deb:

![image](https://github.com/user-attachments/assets/660e61de-9705-41a0-9fa1-51b73d5bd48a)

**Comment by leon3s** [comment]
I guess extending the dependencies will fix that i'll add it in 0.1.4 release

**Comment by leon3s** [comment]
0.1.4 is release and should fix the problem

**Comment by melroy89** [comment]
Well not this `deb` file installation error:

![image](https://github.com/user-attachments/assets/7f57982d-5e25-4dc8-aca6-286932a34621)

**Comment by melroy89** [comment]
So under Ubuntu 24.04 this package is now called: `libwebkit2gtk-4.1-0`.  https://packages.ubuntu.com/noble/libwebkit2gtk-4.1-0

**Comment by melroy89** [comment]
So in the package depends for Ubuntu you need to use an "OR" syntax basically like : `libwebkit2gtk-4.0-37 | libwebkit2gtk-4.1-0`. So on older Ubuntu releases it might need to install and find the 4.0-37 version, while on newer ubuntu releases it's able to find and install 4.1.0 of this package dependency. 

I hope that is possible to do that in Nextauri, I use CMake and with CMake I can use this `|` pipe symbol this purpose.. Here a real-life example of my WineGUI project: https://gitlab.melroy.org/melroy/winegui/-/blob/main/cmake/packaging.cmake?ref_type=heads#L40

**Comment by leon3s** [comment]
I doubt this is possible i'll see what i can do

**Comment by melroy89** [comment]
See also: https://serverfault.com/questions/1055356/how-to-add-multiple-dependencies-to-debian-package

**Comment by osering** [comment]
Lubuntu 24.04.01 v0.1.5. also has unsatisfiable dependencies. 
instead of libjavascriptcoregtk-4.0-18 needs 4.1.0
instead of libicu70, needs 74
instead of libwebkit2gtk-4.0-37 needs 4.1-0

**Comment by melroy89** [comment]
> On Lubuntu 24.04.01 v0.1.5 .deb also has unsatisfiable dependencies.

Yes well. It's the same issue, since Lubuntu 24.04 is based on Ubuntu 24.04 (Noble Numbat).

**Comment by leon3s** [comment]
I'll create a build workflow that run on ubuntu 24 to fix this issue
